src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -590,10 +590,12 @@ bool primitive_inst::update_impl() {
                                 return;
                         }
 
-                        auto impl = _node->type()->choose_impl(*_node, updated_params_no_dyn_pad);
                         if (!can_be_optimized()) {
-                            auto kernels = _program->get_kernels_cache().compile(updated_params_no_dyn_pad, impl->get_kernels_source());
-                            impl->set_kernels(kernels);
+                            auto impl = _node->type()->choose_impl(*_node, updated_params_no_dyn_pad);
+                            if (impl->get_kernels_source().size() > 0) {
```

Comment:
when impl->get_kernels_source().size() == 0  though it is not can_be_optimized?

Reply:
This is the case of onednn_impl.